docs/linters.md

```diff
@@ -248,6 +249,39 @@ It will suggest to remove the pointer from the field, and update the `json` tag
 If you prefer not to suggest fixes for pointers in required fields, you can change the `pointerPolicy` to `Warn`.
 The linter will then only suggest to remove the `omitempty` value from the `json` tag.
 
+## SSATags
+
+The `ssatags` linter ensures that all array fields in Kubernetes API objects have the appropriate
+listType markers (atomic, set, or map) for proper Server-Side Apply behavior.
+
+This linter helps enforce proper Server-Side Apply (SSA) semantics by ensuring array fields
+are correctly configured with the right listType markers. Proper SSA configuration prevents
+data corruption and ensures predictable merge behavior when applying configuration changes.
+
+The linter checks for:
+
+1. Missing listType markers on array fields
+2. Invalid listType values (must be atomic, set, or map)
+3. Missing listMapKey markers for listType=map arrays
+
+### Configuration
+
+```yaml
+lintersConfig:
+  ssaTags:
+    listTypeSetUsage: Warn | SuggestFix | Ignore # The policy for listType=set usage. Defaults to `Warn`.
+```
+
+### Fixes
+
+The `ssatags` linter can automatically suggest fixes for missing or invalid listType markers.
+
+When a listType=set is detected, the linter can suggest changing it.
+
+- `Warn`: Emit a warning when listType=set is used
+- `SuggestFix`: Emit a warning and remove listType=set marker
```

Comment:
If we remove it, then the rule is violated again, so, what are we suggesting here? Should users have to then re-add atomic or map?

Reply:
I totally agree with you. Let's remove the SuggestFix entirely.